### DISCUSSION TITLE
Apex Test code coverage - make colors configurable

### DISCUSSION THREAD

**Discussion by AndrewStopchenko-SO** [forcedotcom/salesforcedx-vscode#5807] [Ideas]
**Is your feature request related to a problem? Please describe.**
It is slightly inconvenient (to me) that the Apex code coverage default highlighting colors are lime/red.
When there are multiple `if () {}` statements in the code and the unit test covers only part of them, it is somewhat hard to detect the non-covered lines when the green and red highlighting colors are mixed back to back (see **example1** below).

I find it much easier to spot the uncovered lines with a fast glance e.g. when the "blue"/"red" palette is used, like it is in the Developer Console.

**Describe the solution you'd like**
If possible, please expose the covered/non-covered colors as a VS Code Apex extension settings.
It should be fine if "lime"/"red" highlight colors are used by default (if everybody are fine with that), but the end users could override those colors via settings.

**Describe alternatives you've considered**
I have found the JS file which handles test coverage decoration (_%user%\.vscode\extensions\salesforce.salesforcedx-vscode-apex-53.1.0\out\src\codecoverage\decorations.js_) and updated it to highlight the covered lines with blue (please see **example2** below).
Unfortunately, this "local" override gets removed every time the extension is updated.

**Additional context**
**Example1** - current default color (lime/red):
![image](https://user-images.githubusercontent.com/63401193/139414784-a9279d9a-7a80-484b-b103-1eaf71f16f76.png)

**Example2** - locally overridden color (blue/red):
![image](https://user-images.githubusercontent.com/63401193/139415115-4d86586f-79b4-49cc-99d9-98c71d045abb.png)
_Please note: the blue color `rgba(....)` values should be considered not to be mixed up with the text selection._

**Comment by andrew-chiz-muzychuk** [comment]
As a colorblinded person I need this.

**Comment by RupertBarrow** [comment]
I'm colour-blind, I need this.
12.5% of men and 0.5% of women are colour-blind : please help us !

**Comment by AndrewStopchenko-SO** [comment]
Implemented in #6147.

https://github.com/forcedotcom/salesforcedx-vscode/issues/6147#issuecomment-2805610465:

> Now users can modify their VSCode's User/Workspace Settings, by providing `workbench.colorCustomizations` object with `testing.coveredBackground` and `testing.uncoveredBackground` color values.
> 
> By default, VSCode uses yellow color for covered lines background in standard dark/white Color Themes.
> 
> An example to mark covered lines with blue:
> ```json
>   "workbench.colorCustomizations": {
>     "testing.coveredBackground": "#2d79ff50",
>   },
> ```
